FreeRTOS/Test/CMock/tasks/tasks_1_utest.c

```diff
@@ -3550,6 +3550,29 @@ void test_vTask_Set_Get_ThreadLocalStoragePointer_success( void )
     TEST_ASSERT_EQUAL_PTR( pValue, ret_pValue );
 }
 
+/* this test ensures that if the index is negative, then NULL is returned. */
+void test_vTask_Set_Get_ThreadLocalStoragePointer_successNegative( void )
+{
+    TaskHandle_t task_handle;
+    void * ret_pValue;
+    int i = 0xABCD;
+    void * value = &i;
+
+    /* Setup */
+    create_task_priority = 3;
+    task_handle = create_task();
+    /* Expectations */
+    /* API Call */
+
+    #pragma GCC diagnostic ignored "-Warray-bounds"
+    *( task_handle->pvThreadLocalStoragePointers - 1 ) = value;
```

Comment:
Not clear to me that this line is a good idea since you're going to corrupt part of the TCB. Maybe instead your verification step should be a short loop that scans backwards through enough negative indices that you know you're reading past the beginning of the TCB. The preceding fields won't all be zero, so you know you won't be getting a false negative.

Alternatively you could use xTaskCreateStatic to place the TCB at the end of a buffer that will be initially filled with a known, non-NULL, pattern.

Or just get rid of this check if we're sure the underflow will be detected by other tools.